client/src/components/FlightSearch.js

```diff
@@ -73,81 +82,69 @@ const FlightSearch = ({ submit }) => {
-  	if (from && to && departureDate && returnDate) {
-  		if (returnDate >= departureDate) {
-  			const response = await fetch(
-  				`api/flights/quotes/${fromCity}/${toCity}/${formattedDepartureDate}/?inboundDate=${formattedReturnDate}`
-  			);
-  			const data = await response.json();
-  			if (data.status === 200) {
-  				submit({
-  					date: departureDate.toString(),
-  					flights: data,
-  				});
-
-				  console.log(response)
-  			} else if ("from" in data) {
-  				setFromError(true);
-  			} else if ("to" in data) {
-  				setToError(true);
-  			} else if ("outboundDate" in data) {
-  				setDepartDateError(true);
-  			} else {
-  				setReturnDateError(true);
-  			}
-  		} else {
-  			setReturnDateError(true);
-  		}
-  		// return date is optional
-  	} else if (from && to && departureDate) {
-  		const response = await fetch(
-  			`api/flights/quotes/${from}/${to}/${formattedDepartureDate}`
-  		);
-  		const data = await response.json();
-  		if (data.status === 200) {
-  			submit({
-  				date: departureDate.toString(),
-  				flights: data,
-  			});
-
-			  console.log(data)
-  		} else if ("from" in data) {
-  			setFromError(true);
-  		} else if ("to" in data) {
-  			setToError(true);
-  		} else if ("outboundDate" in data) {
-  			setDepartDateError(true);
-  		} else {
-  			setReturnDateError(true);
-  		}
-  	} else if (!from) {
-  		setFromError(true);
-  	} else if (!to) {
-  		setToError(true);
-  	}
-  };
- 
+		event.preventDefault();
+		const fromCity = cities.find((city) => from === city.title).title;
+		const toCity = cities.find((city) => to === city.title).title;
+		const formattedDepartureDate = formatDate(departureDate);
 
- 
+		if (from && to && from !== to && departureDate && returnDate) {
+			const formattedReturnDate = formatDate(returnDate);
+			if (returnDate >= departureDate) {
+				const response = await fetch(
+					`api/flights/quotes/${fromCity}/${toCity}/${formattedDepartureDate}/?inboundDate=${formattedReturnDate}`
+				);
+				const data = await response.json();
+				if (response.status === 200) {
+					submit({ flights: data });
+					console.log(data);
+				} else if ("from" in data) {
+					setFromError(true);
+				} else if ("to" in data) {
+					setToError(true);
+				} else if ("outboundDate" in data) {
+					setDepartDateError(true);
+				} else if ("inboundDate" in data) {
+					setReturnDateError(true);
+				}
+			} else {
+				setReturnDateError(true);
+			}
+			// return date is optional
+		} else if (from && to && from !== to && departureDate) {
+			const response = await fetch(
+				`api/flights/quotes/${fromCity}/${toCity}/${formattedDepartureDate}`
+			);
+			const data = await response.json();
+			if (response.status === 200) {
+				submit({ flights: data });
+			} else if ("from" in data) {
+				setFromError(true);
+			} else if ("to" in data) {
+				setToError(true);
+			} else if ("outboundDate" in data) {
+				setDepartDateError(true);
+			}
```

Comment:
There's quite a bit of duplicated code here. Check to see if you can reduce it 🙂

Reply:
This is my code. I found the way to reduce it was to isolate the fetch call into its own function. Please see below: 

	const handleFetch = async () => {
		const fromCity = cities.find((city) => from === city.title).title;
		const toCity = cities.find((city) => to === city.title).title;
		const formattedDepartureDate = formatDate(departureDate);
		const formattedReturnDate = returnDate ? formatDate(returnDate) : null;
		const response = await fetch(
			`api/flights/quotes/${fromCity}/${toCity}/${formattedDepartureDate}${
				returnDate ? `/?inboundDate=${formattedReturnDate}` : ""
			}`
		);
		const data = await response.json();
		if (response.status === 200) {
			submit({ flights: data });
			console.log(data);
		} else if ("from" in data) {
			setFromError(true);
		} else if ("to" in data) {
			setToError(true);
		} else if ("outboundDate" in data) {
			setDepartDateError(true);
		} else if ("inboundDate" in data) {
			setReturnDateError(true);
		}
	};
	const handleSubmit = (event) => {
		event.preventDefault();
		if (from && to && from !== to && departureDate && returnDate) {
			if (returnDate >= departureDate) {
				handleFetch();
			} else {
				setReturnDateError(true);
			}
			// return date is optional
		} else if (from && to && from !== to && departureDate) {
			handleFetch();
		} else if (from === to || !to) {
			setToError(true);
		} else if (!from) {
			setFromError(true);
		}
	};

Reply:
Thanks @newCodeWriter.